### PR TITLE
Linear clean

### PR DESCRIPTION
**Pull request by anguta**
Implementation of linear clean for use in analysis of visibility data. The sinc_downweight method will eventually be referenced by hera_pspec (and the version therein deprecated). 

I've also added a (hopefully?) helpful tutorial notebook in "examples". 

cheers,

-Aaron

### CODE REVIEW

File: uvtools/dspec.py [hunk 1]
```diff
@@ -775,3 +868,71 @@ def delay_filter_leastsq(data, flags, sigma, nmax, add_noise=False,
...
+        float or list of floats of widths of delay filter windows in nanosec
+    filter_factors: float or list
+        float or list of floats of filtering factors.
+    cache: dictionary, optional dictionary storing filter matrices with keys
+    (nchan, df, ) + (filter_centers) + (filter_widths) + \
+    (filter_factors)
+
+    !!!-------------
+    WARNING: The following parameters are intended for studying weighting
+    but shouldn't not be changed from defaults in practical data analysis!
```
Comment (nkern) [inline uvtools/dspec.py]: typo: shouldn't not be

File: uvtools/dspec.py [hunk 2]
```diff
@@ -775,3 +868,71 @@ def delay_filter_leastsq(data, flags, sigma, nmax, add_noise=False,
         mdl_array[i, :] = bf_model
 
     return mdl_array, cn_array, inp_data
+
+
+def sinc_downweight_mat_inv(nchan, df, filter_centers, filter_widths,
+                            filter_factors, cache = {}, wrap = False, wrap_interval=1,
+                            nwraps = 1000, no_regularization = False):
+    """
+    Computes inverse of clean weights for a baseline.
```
Comment (nkern) [inline uvtools/dspec.py]: did we say we weren't referencing "clean", and is this a holdout of that? I forget

Comment (anguta) [reply uvtools/dspec.py]: you're right. I'm removing the reference.

File: uvtools/dspec.py (hunk 1 above)
Comment (anguta) [reply uvtools/dspec.py]: Thanks for catching this. I've fixed it.

**Pull request merged**